lib/datadog/statsd/serialization/stat_serializer.rb

```diff
@@ -38,16 +38,17 @@ def global_tags
         attr_reader :tag_serializer
 
         def formated_name(name)
-          if name.is_a?(String)
-            # DEV: gsub is faster than dup.gsub!
-            formated = name.gsub('::', '.')
+          formated = name.is_a?(String) ? name : name.to_s
```

Comment:
`String#to_s` returns self, so this should be equivalent to

```suggestion
          formated = name.to_s
```


```
>> foo = "foo"
=> "foo"
>> foo.object_id
=> 17120
>> foo.to_s.object_id
=> 17120
```

(also formated is a typo `s/formated/formatted` 😅)

Reply:
> String#to_s returns self, so this should be equivalent to

changed in https://github.com/DataDog/dogstatsd-ruby/pull/294/commits/158507b35cc9de7df62bf4d0305e70af1b0f7fdc

> (also formated is a typo s/formated/formatted 😅)

thanks for calling that out!

> If we want to reduce allocations further, having a Symbol here is probably the most common non-string case, and 
Symbol#to_s allocates whereas Symbol#name does not.

so we'd change the call to `name.is_a?(Symbol) ? name.name : name.to_s` (and rename `name` to e.g. `metric_name`)?

Reply:
Yes! I slightly prefer `Symbol === name` instead of `is_a?`, since a constant receiver class should be a bit friendlier to method caches, but I don't think it matters too much.